tests/ImageSharp.Tests/Processing/Processors/Dithering/DitherTests.cs

```diff
@@ -155,7 +156,7 @@ public void DitherFilter_ShouldNotDependOnSinglePixelType<TPixel>(TestImageProvi
                 appendPixelTypeToFileName: false);
         }
 
-        [Theory]
+        [Theory(Skip = "Unable to assign capacity smaller than the image.")]
```

Comment:
@antonfirsov This is problematic. The maximum height of the image in rows provides less of a buffer than is required by the cache.

Reply:
I think I should just drop pooling here. I wrote code to workaround our pooler limitations using `ArrayPool<byte>.Shared` but that's gonna just allocate the buffer anyway since we just exceed the maximum buffer amount.

Reply:
Scratch that. After reading the [source](https://source.dot.net/#System.Private.CoreLib/TlsOverPerCoreLockedStacksArrayPool.cs) I realized that it's not pre-filling multiple buckets but filling a bucket on demand. Using `ArrayPool<short>.Shared` seems appropriate for this use case.

Reply:
You had that right the first time.  The comment [here](https://source.dot.net/#System.Private.CoreLib/TlsOverPerCoreLockedStacksArrayPool.cs,32) is weird because `NumBuckets` is the length of the buckets array, so max index is one less than that, which is the bucket for 1MiB arrays.  Anything larger ends up with an invalid bucket index and will new up an array.

Reply:
Yeah but I was using `byte` and casting to avoid using a pool that I couldn't find usage of in the runtime. Now I'm using `short` my buffer length is less so it get pooled. The index is determined purely by the minimum length required.

Reply:
Ohhhh, gotcha :)